packages/web/pages/user/my-account/my-orders.js

```diff
@@ -0,0 +1,348 @@
+														onClick={() =>
+															setCurrentOrderMessages(order)
+														}
+													>
+														<FiMessageSquare />
+													</IconButton>
+													<IconButton
+														variant="remove"
+														aria-label="Cancel order"
+														disabled={
+															status === dealStatusEnum.DEAL_CANCELLED
+														}
+														onClick={() => openModal('cancelOrder')}
+													>
+														<FiX />
+													</IconButton>
+												</DealActions>
+											),
+										};
+									})}
+									hideItemsByKey={['id']}
+									currentPage={currentPage}
+									totalPages={totalPages}
+									totalItems={totalItems}
+									itemsPerPage={itemsPerPage}
+									currentOrder={currentSort.order}
+									sortOptions={sortOptions}
+									handlePagination={handlePagination}
+									handleSortBy={handleSortBy}
+									enablePagination
+								/>
+							) : (
+								<NoOrders>{t('account:messages.noOrdersToShow')}</NoOrders>
+							)}
+						</MainContent>
+					</MainContentContainer>
+				)}
+			</Protected>
+		</Container>
+	);
+};
+
+MyOrders.propTypes = {
+	orders: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+	currentPage: PropTypes.number.isRequired,
+	totalPages: PropTypes.number.isRequired,
+	totalItems: PropTypes.number.isRequired,
+	currentSort: PropTypes.shape({
+		by: PropTypes.string,
+		order: PropTypes.string,
+	}),
+};
+
+MyOrders.defaultProps = {
+	currentSort: {},
+};
+
+MyOrders.getInitialProps = async (ctx) => {
+	const { query } = ctx;
+
+	const page = Number(query.page) || 1;
+
+	return {
+		orders: ordersMock,
+		currentPage: page,
+		totalPages: 1,
+		totalItems: 3,
+		currentSort: { by: query.orderBy, order: query.order },
+		sortOptions,
+		namespacesRequired: ['helper', 'account', 'profile', 'datagrid'],
+	};
+};
+
+export const Container = styled.div`
+	display: flex;
+	margin: 0 auto;
+	background-color: ${({ theme }) => theme.colors.whiteSmoke};
+	padding: 3rem 4rem 6rem;
+
+	> section:first-child {
+		margin-right: 4rem;
+	}
+
+	@media screen and (max-width: 950px) {
+		flex-direction: column;
+
+		> section:first-child {
+			margin-bottom: 1rem;
+		}
+	}
+`;
+
+export const MainContentContainer = styled.section`
+	width: 100%;
+`;
+
+export const MainContent = styled.div`
+	min-height: 80vh;
+	background-color: ${({ theme }) => theme.colors.white};
+	padding: 2rem;
+`;
+
+export const InfoContainer = styled.div`
+	display: flex;
+	align-items: center;
+	justify-content: space-between;
+	margin-bottom: 1rem;
+
+	@media screen and (max-width: 950px) {
+		flex-direction: column;
+
+		button {
+			margin-bottom: 1rem;
+		}
+	}
+`;
+
+export const NoOrders = styled.span`
+	color: ${({ theme }) => theme.colors.darkGray};
+	font-size: 2rem;
+`;
+
+const statusModifiers = {
+	dealStruck: (colors) => css`
+		color: ${colors.secondary};
+		&::before {
+			background: ${colors.secondary};
+		}
+	`,
```

Comment:
See the comment above ☝️. Apply the same to other properties of this object.
```suggestion
const statusModifiers = {
	[dealStatusEnum.DEAL_STRUCK]: (colors) => css`
		color: ${colors.secondary};
		&::before {
			background: ${colors.secondary};
		}
	`,
```

Reply:
We need to check for the actual status, otherwise it will call all methods this way: `${statusModifiers[dealStatusEnum.DEAL_STRUCK](colors)};`

Reply:
See the comment above ☝️